crates/nu-cmd-base/src/arg_glob.rs

```diff
@@ -0,0 +1,115 @@
+// utilities for expanding globs in command arguments
+use nu_protocol::{ShellError, Spanned};
+use std::path::PathBuf;
+use wax::{Glob, LinkBehavior, WalkBehavior};
+
+// standard glob options to use for filesystem command arguments
+const FS_CMD_WALK_BEHAVIOR: WalkBehavior = WalkBehavior {
+    depth: usize::MAX,
+    link: LinkBehavior::ReadTarget,
+};
+
+// expand a glob into matching paths, using FS_CMD_WALK_BEHAVIOR
+//todo: see about returning WalkError or BuildError, or enabling miette and dealing with diagnostic
+pub fn arg_glob(
+    pattern: &Spanned<String>, // the glob
+    include_dirs: bool,        // include dirs in results.  Default (f) only include files
+    cwd: &PathBuf,             // current working directory
+) -> Result<Vec<PathBuf>, ShellError> {
+    // eventually, windows only, and with better performance
+    // transform something like: C:\Users\me\.../{abc,def}*.txt to C\:/Users/me/.../{abc,def}*.txt
+    // such a string has good chance of working as a rooted glob pattern on Windows
+    // User will have to quote metachars with [c] rather than \c.
+    let processed_pattern = pattern
+        .item
+        .replace('\\', "/")
+        .replace(r#":/"#, r#"\:/"#) // must come after above
+        .replace('(', r#"\("#) // when path includes Program Files (x86)
+        .replace(')', r#"\)"#);
```

Comment:
I'm wondering what effect this will have on paths with spaces like `Program Files` and `Program Files (x86)`. I see that lots of things are being replaced here but I'm not sure how that affects things here with these particular paths.

Reply:
Embedded spaces won't be affected by these changes.  A pattern or simple path with an embedded space would have to be enclosed in quotes anyway.  It has been tested and works with `C:\Program Files (x86)\*`!